views/js/controller/list/index.js

```diff
@@ -92,19 +93,41 @@ define([
     function handleEditList (targetUri) {
         const uri = getUriValue(targetUri);
         const $listContainer = findListContainer(uri);
-        const offset = $listContainer.find('ol').children('[id^=list-element]').length;
+        const list = $listContainer.find('ol');
+        const offset = list.children('[id^=list-element]').length;
+        let totalItems = 0;
+
+        function isLimitReached() {
+            let maxItems = 1000;
+
+            if(list.data("max-items") > 0) {
+                maxItems = list.data("max-items");
+            }
+
+            return totalItems >= maxItems;
```

Comment:
Maybe we could simplify? https://api.jquery.com/data/ assuming that we rely in this data coming from BE?

```suggestion
            return list.data("max-items", 1000) >= maxItems;
```

Reply:
Hmmm... `list.data('max-items', 1000);` seems to set the `max-items` data attribute value in the DOM to `1000` (i.e. it is not the default value returned -- two parameters makes it work as a setter).

Reply:
Maybe this instead? 

```suggestion
            const maxItems = list.data("max-items") > 0 ? list.data("max-items") : 1000;

            return totalItems >= maxItems;
```

Reply:
Even though I like this there seems to be a best practice rule about not using ternary operators 😢

Reply:
You are right 🤦. I read it too fast. I am just wondering we are validating if `> 0`, but we do not check if it is not a number. 

It seems we do not rely on BE value, but we do not do all the checks. I think it is unlikely we will pass 0 to the BE, so I am not sure we are overvalidating again...

Reply:
In this case we are not directly relying on the backend but on a config parameter. Theoretically we could set the env var to `0` (or a negative value) and it would pass the `ContainerServiceProviderInterface` `int()` validation. That's the reason for checking if is greater than zero and use the default value in that case. 

Overly defensive programming, sure, but I think it can be used as a feature as well: Set the env var to 0 (or a negative number) and let TAO use a default automatically (1K in this case).

I'm not testing if `max-items` is a number because, if it isn't, `list.data("max-items") > 0` returns `false` (just checked with a string attribute in the dev console, for example, `$('#browser-check').attr('class') > 0` is `false`).

Reply:
Is it not enough to make sure it is not 0 in the BE? Cause this template is built there.

Reply:
I am closing since this is optional, but it was a nice discussion :)